File: dotnet/SpiderEngine/Abstract/ISpiderHttpClient.cs
namespace SpiderEngine.Abstract;

using SpiderEngine.Model;

public interface ISpiderHttpClient
{
    Task<(string Content, IDictionary<string, string> Headers)> GetAsync(string url, IDictionary<string, string> headers, int success = 200, bool redirect = true);
    Task<(string Content, IDictionary<string, string> Headers)> PostAsync(string url, IDictionary<string, string> headers, string payload, SpiderPayloadType type, int success = 200, bool redirect = true);
}

public interface ISpiderHttpClientProvider
{
    ISpiderHttpClient Create(EngineOptions options);
}

File: android/src/main/kotlin/com/itoken/team/wutspider/interfaces/ISpiderHttpClient.kt
package com.itoken.team.wutspider.interfaces

import com.itoken.team.wutspider.model.SpiderPayloadType


interface ISpiderHttpClient {

    fun get(url: String, headers: Map<String, String>, success: Int = 200): HttpClientResponse

    fun post(
        url: String,
        headers: Map<String, String>,
        payload: String,
        payloadType: SpiderPayloadType,
        success: Int = 200
    ): HttpClientResponse

    data class HttpClientResponse(
        val statusCode: Int,
        val headers: Map<String, String>,
        val content: String
    )

}